stl/inc/condition_variable

```diff
@@ -92,8 +110,8 @@ public:
     template <class _Lock, class _Rep, class _Period>
     cv_status wait_for(_Lock& _Lck, const chrono::duration<_Rep, _Period>& _Rel_time) { // wait for duration
         if (_Rel_time <= chrono::duration<_Rep, _Period>::zero()) {
-            _Lck.unlock();
-            _Relock(_Lck);
+            _Unlock_guard<_Lock> _Unlock_outer{_Lck};
+            (void) _Unlock_outer;

```

Comment:
Why this discarded `void` cast? Did a compiler complain about this being unused?

Reply:
I'm being paranoid about https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-4-c4189?view=vs-2019